aw_watcher_window/main.py

```diff
@@ -82,59 +39,23 @@ def main():
-
-            # If windows are not the same, insert it as a new event
-            elif last_window != current_window:
-                if last_window is not None:
-                    # Create last_window event
-                    duration = now - last_window_start
-                    labels = ["title:" + last_window["title"]]
-                    labels.append("appname:" + last_window["appname"])
-                    last_window_event = Event(label=labels, timestamp=last_window_start, duration=duration)
-                    # Send last_window event
-                    client.replace_last_event(bucketname, last_window_event)
-                    # Log
-                    logger.debug("Window is no longer active: " + str(last_window))
-                    logger.debug("Duration: {}s".format(str(duration.total_seconds())))
-
-                # Create current_window event
-                duration = timedelta()
-                labels = ["title:" + current_window["title"]]
-                labels.append("appname:" + current_window["appname"])
-                current_window_event = Event(label=labels, timestamp=now, duration=duration)
-                # Send events
-                client.send_event(bucketname, current_window_event)
-                last_event_time = now
-
-                # Log
-                logger.info("Window became active: " + str(current_window))
-                # Store current window
-                last_window = current_window
-                last_window_start = now
-
-            # If windows are the same and update_time has passed (default 15sec), replace last event with this event for updated duration
-            elif now - timedelta(seconds=update_time) > last_event_time:
-                # Create current_window event
-                duration = now - last_window_start
-                labels = ["title:" + current_window["title"]]
-                labels.append("appname:" + current_window["appname"])
-                current_window_event = Event(label=labels, timestamp=last_window_start, duration=duration)
-
-                # Send current_window event
-                client.replace_last_event(bucketname, current_window_event)
-                last_event_time = now
-
         except Exception as e:
             logger.error("Exception thrown while trying to get active window: {}".format(e))
             traceback.print_exc(e)
-        sleep(poll_time)
+            continue
+
+        now = datetime.now(timezone.utc)
+        if current_window is None:
+            logger.debug('Unable to fetch window, trying again on next poll')
+        else:
+            # Create current_window event
+            labels = ["title:" + current_window["title"]]
+            labels.append("appname:" + current_window["appname"])
+            current_window_event = Event(label=labels, timestamp=now)
+
+            client.heartbeat(bucketname, current_window_event, 2*args.poll_time)
```

Comment:
One second. The `2*poll_time` is there since the loop block obviously takes more than `poll_time`seconds (we know it takes at least that long due to the `sleep(poll_time)`). 

A better value would probably be `poll_time + 1`, I'll fix that tomorrow.